tests/test_runlib.py

```diff
@@ -642,6 +720,45 @@ def test_create_metadata_with_expected_product(self):
     self.assertEqual(list(link.signed.products.keys()), [self.test_product])
     os.remove(self.link_name)
 
+  def test_create_metadata_with_expected_product_in_target_directory(self):
+    """Test record stop records expected product in target directory """
+    try:
+      tmp_dir = tempfile.mkdtemp()
+      in_toto_record_start(self.step_name, [], self.key)
+      in_toto_record_stop(self.step_name, [self.test_product], self.key,
+          metadata_directory=tmp_dir)
+      link_path = os.path.join(tmp_dir, self.link_name)
+      link = Metablock.load(link_path)
+      self.assertEqual(list(link.signed.products.keys()), [self.test_product])
```

Comment:
I'm not sure we're very concerned about the products here.

Reply:
I am thinking that:
If we already compare the dumped link with the metadata directory is equal to the dumped link without the metadata directory, we don't need this function here to check for the products.
Or maybe we should check something other than products?
What do you think?